app/models/submission.rb

```diff
@@ -63,7 +63,7 @@ def correction_size
   end
 
   def preceding_tutorial(user)
-    assignment.previous&.map { |a| a.tutorial(user) }&.compact&.first
+    assignment.previous&.filter_map { |a| a.tutorial(user) }&.first
```

Comment:
Reviewing the code, I'm not sure why the autocorrection of this cop is not marked as safe. The justification is not too clear in my opinion:

> This cop’s autocorrection is unsafe because `map { …​ }.compact` that is not compatible with `filter_map`. ~ from [here](https://docs.rubocop.org/rubocop-performance/cops_performance.html#safety-11)

Having looked at the API for [`compact`](https://apidock.com/ruby/v2_5_5/Array/compact) (remove `nil` elements from array) and [`filter_map`](https://ruby-doc.org/3.1.4/Enumerable.html#method-i-filter_map):

> Returns an array containing truthy elements returned by the block. With a block given, calls the block with successive elements; returns an array containing each truthy value returned by the block:

it seems like this cop would be safe. But it's not, so I'm just wondering what edges cases there might be and if it's really safe to accept this change.

Reply:
I do not see why it shouldn't be safe in our situation (when I reviewed the code I studied the same API that you linked, looked at sveral blogs where examples where given and played around a little in the Rails console).

Reply:
Ah, I found [this issue](https://github.com/rubocop/rubocop-performance/issues/280), which is a case where it gives different results, but this should not affect us here.

Reply:
I've opened https://github.com/rubocop/rubocop-performance/pull/450 to improve the wording in the docs.

Reply:
You may resolve this conversation.